site/src/components/pages/hub/BlockDataContainer.tsx

```diff
@@ -29,31 +38,78 @@ export const BlockDataContainer: VoidFunctionComponent<
   BlockDataContainerProps
 > = ({ metadata, schema, blockModule }) => {
   const [blockDataTab, setBlockDataTab] = useState(0);
-  const [blockTab, setBlockTab] = useState(0);
+  const [blockVariantsTab, setBlockVariantsTab] = useState(0);
   const [blockModalOpen, setBlockModalOpen] = useState(false);
+  const [alertSnackBarOpen, setAlertSnackBarOpen] = useState(false);
 
-  const example = metadata.examples?.[0];
+  const [text, setText] = useState("{}");
 
-  const [text, setText] = useState(
-    example ? JSON.stringify(example, undefined, 2) : "",
-  );
+  const previousBlockVariantsTab = useRef(-1);
+  const propertiesToRemove = useRef<string[]>([]);
```

Comment:
I think we can implement the intended feature without this ref

Reply:
`previousBlockVariantsTab` is required, but the requirement of `propertiesToRemove` depends on the resolution for this comment: https://github.com/blockprotocol/blockprotocol/pull/202#discussion_r810792505.

CC: @CiaranMn